ansible_base/activitystream/signals.py

```diff
@@ -1,11 +1,36 @@
 import logging
+import threading
+from contextlib import contextmanager
 
 from ansible_base.lib.utils.models import current_user_or_system_user, diff
 
 logger = logging.getLogger('ansible_base.activitystream.signals')
 
 
+class ActivityStreamEnabled(threading.local):
+    def __init__(self):
+        self.enabled = True
+
+    def __bool__(self):
+        return self.enabled
+
+
+activitystream_enabled = ActivityStreamEnabled()
+
+
+@contextmanager
+def no_activity_stream():
+    activitystream_enabled.enabled = False
```

Comment:
I do kind of like that AWX maintained a `prior_value` store here. It's not that you'd ever switch True-False-True-False, but you would logically want a True-False-False nesting

```python
with no_activity_stream():
  with no_activity_stream():
    pass
  obj.save()
```

In this case, I would still want `obj.save()` to leave no activity stream entry. This is perfectly testable and all.